test/uk/gov/hmrc/hec/controllers/TaxCheckControllerWithInternalAuthEnabledSpec.scala

```diff
@@ -284,23 +322,33 @@ class TaxCheckControllerSpec extends ControllerSpec with AuthSupport {
       "return a 400 (bad request)" when {
 
         "the JSON in the request cannot be parsed" in {
-          status(performAActionWithJsonBody(JsString("hi"))) shouldBe BAD_REQUEST
-
+          mockInternalAuth(expectedPredicate)(Future.successful(Unit))
+          status(performAActionWithJsonBodyAndHeader(JsString("hi"))) shouldBe BAD_REQUEST
         }
 
       }
 
       "return an 500 (internal server error)" when {
 
         "there is an error saving the tax check" in {
+          mockInternalAuth(expectedPredicate)(Future.successful(Unit))
           mockMatchTaxCheck(individualMatchRequest)(Left(Error(new Exception("Oh no!"))))
 
-          val result = performAActionWithJsonBody(Json.toJson(individualMatchRequest))
+          val result = performAActionWithJsonBodyAndHeader(Json.toJson(individualMatchRequest))
           status(result) shouldBe INTERNAL_SERVER_ERROR
         }
 
       }
 
+      "return a 401 Unauthorised" when {
+
+        "there is no authenticated session" in {
+          mockInternalAuth(expectedPredicate)(Future.failed(InvalidBearerToken("Internal Auth token")))
+          assertThrows[RuntimeException](await(performAActionWithJsonBodyAndHeader(Json.toJson(companyMatchRequest))))
```

Comment:
this test shows an exception being thrown instead of a 401 being returned - does that mean we won't actually get a 401 back?

Reply:
When i checked on postman, no token or invalid token gives 401 , so i started with 401 then realised it throwing an unAuthorised Exception in here

Reply:
having a closer dig, you may need to fail the `Future` with `UpstreamErrorResponse` which contains a `401` status code in it for us to return a 401:
https://github.com/hmrc/internal-auth-client/blob/main/internal-auth-client-play-26/src/main/scala/uk/gov/hmrc/internalauth/client/actions/AuthActionBuilder.scala#L37

Reply:
thanks, let me try this